### PR TITLE
feat(init): use jsr specifier for @std/assert

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit changes "deno init" subcommand to use "jsr:" specifier for
standard library "assert" module. It is unversioned, but we will change it
to `@^1` once `@std/assert` release version 1.0.

This allows us to start decoupling `deno` and `deno_std` release. The
release scripts have been updated to take that into account.

### CODE REVIEW

File: tests/integration/init_tests.rs [hunk 1]
```diff
@@ -6,7 +6,10 @@ use util::TestContextBuilder;
 
 #[test]
 fn init_subcommand_without_dir() {
-  let context = TestContextBuilder::new().use_temp_cwd().build();
+  let context = TestContextBuilder::new()
+    .env("JSR_URL", "https://jsr.io")
+    .use_temp_cwd()
+    .build();
```
Comment (bartlomieju) [inline tests/integration/init_tests.rs:12]: Not great, but I don't have a better idea how to test it properly. Maybe it's fine?

Comment (dsherret) [reply tests/integration/init_tests.rs:12]: Add an `@std/path` to the test server?

File: cli/tools/init/templates/main_test.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { assertEquals } from "{CURRENT_STD_URL}assert/mod.ts";
+import { assertEquals } from "jsr:@std/assert";
```
Comment (dsherret) [inline cli/tools/init/templates/main_test.ts:1]: Can we use a deno.json import entry for `@std/assert` and have a version range specified in there? Maybe we could just reuse the deno add code for now

Comment (bartlomieju) [reply cli/tools/init/templates/main_test.ts:1]: Er, I'm not sure I understand - I can certainly move it to `deno.json`, but whole point of this PR was to not require bumping version on each release. I think we can just leave it unversioned for the time being and change it to `@^1` once we release appropriate version for `@std/assert`?

How `deno add` code is relevant here?

Comment (dsherret) [reply cli/tools/init/templates/main_test.ts:1]: Sure, that sounds good to wait until 1.0

File: tests/integration/init_tests.rs (hunk 1 above)
Comment (bartlomieju) [reply tests/integration/init_tests.rs:12]: I would have to dump serveral thousands lines of code into the test server 😬 is it worth it?

Comment (dsherret) [reply tests/integration/init_tests.rs:12]: It doesn't need to align with what's actually on the server. We can just do one file with a couple lines.

**Review by dsherret — APPROVED** [review]
LGTM!

**Pull request merged**